wandb/data_types.py

```diff
@@ -38,53 +92,145 @@ def val_to_json(key, val, mode="summary", step=None):
+            DataFrame here.
+        run (wandb_run.Run): The Run the DataFrame is associated with. We need
+            this because the information we store on the DataFrame is derived
+            from the Run it's in.
+        key (str): Name of the DataFrame, ie. the summary key path in which it's
+            stored. This is for convenience, so people exploring the
+            directory tree can have some idea of what is in the Parquet files.
+        step: History step or "summary".
+
+    Returns:
+        A dict representing the DataFrame that we can store in summaries or
+        histories. This is the format:
+        {
+            '_type': 'data-frame',
+                # Magic field that indicates that this object is a data frame as
+                # opposed to a normal dictionary or anything else.
+            'id': 'asdf',
+                # ID for the data frame that is unique to this Run.
+            'format': 'parquet',
+                # The file format in which the data frame is stored. Currently can
+                # only be Parquet.
+            'project': 'wfeas',
+                # (Current) name of the project that this Run is in. It'd be
+                # better to store the project's ID because we know it'll never
+                # change but we don't have that here. We store this just in
+                # case because we use the project name in identifiers on the
+                # back end.
+            'path': 'media/data_frames/sdlk.parquet',
+                # Path to the Parquet file in the Run directory.
+        }
+    """
+    pandas = util.get_module("pandas")
+    fastparquet = util.get_module("fastparquet")
+    if not pandas or not fastparquet:
+        raise wandb.Error("Failed to save data frame: unable to import either pandas or fastparquet.")
+
+    data_frame_id = util.generate_id()
+
+    df = df.copy()  # we don't want to modify the user's DataFrame instance.
+
+    for col_name, series in df.items():
+        for i, val in enumerate(series):
+            if isinstance(val, WBValue):
+                series.iat[i] = six.text_type(json.dumps(val_to_json(run, key, val, step)))
+
+    # We have to call this wandb_run_id because that name is treated specially by
+    # our filtering code
+    df['wandb_run_id'] = pandas.Series(
+        [six.text_type(run.name)] * len(df.index), index=df.index)
+
+    df['wandb_data_frame_id'] = pandas.Series(
+        [six.text_type(data_frame_id)] * len(df.index), index=df.index)
+    frames_dir = os.path.join(run.dir, DATA_FRAMES_SUBDIR)
+    util.mkdir_exists_ok(frames_dir)
+    path = os.path.join(frames_dir, '{}-{}.parquet'.format(key, data_frame_id))
+    fastparquet.write(path, df)
+
+    return {
+        'id': data_frame_id,
+        '_type': 'data-frame',
+        'format': 'parquet',
+        'project': run.project_name(),  # we don't have the project ID here
+        'entity': run.entity,
+        'run': run.name,
```

Comment:
I think you want run.id here.  If a user overrides the name run.name will be the displayName I know there’s inconsistencies between public and internal.  Let’s chat about how to fix it.

Reply:
Ah, good point.

Reply:
I think I fixed the `id` / `name` / `storage_id` inconsistency a while ago. IIRC I made a `name` property on the data API Run object and an `id` property in the normal Run object. That was a long time ago though so maybe it wasn't as complete of a fix as I remember.

Reply:
Just checked, and it looks like it was fixed but is now broken because of the name/id distinction. I'll change this to `run.id` for now but we'll have to figure out how to fix the inconsistency again.